GitTfs/Core/GitChangeInfo.cs

```diff
@@ -46,15 +47,72 @@ public struct ChangeType
         public static IEnumerable<GitChangeInfo> GetChangedFiles(TextReader reader)
         {
             string line;
+            List<GitChangeInfo> changes = new List<GitChangeInfo>(); ;
             while (null != (line = GetDiffTreeLine(reader)))
             {
                 var change = Parse(line);
 
                 if (FileMode.GitLink == change.NewMode)
                     continue;
 
-                yield return change;
+                changes.Add(change);
             }
+            var filterd = FilterNotRepresentable(changes);
+            return filterd;
+        }
+
+        private static IEnumerable<GitChangeInfo> FilterNotRepresentable(IEnumerable<GitChangeInfo> changed)
+        {
+            //filter case only renames out, they cannot be represented in TFS
+            var remainingChanges = changed.Where(cha => cha.Status != GitChangeInfo.ChangeType.RENAMEEDIT ||
+                String.Compare(cha.path, cha.pathTo, System.StringComparison.OrdinalIgnoreCase) != 0 ||
+                cha.newSha != cha.oldSha).ToList();
+
+            foreach (var cha in remainingChanges)
+            {
+                if(cha.Status == GitChangeInfo.ChangeType.RENAMEEDIT)
+                {
+                    if(String.Compare(cha.path, cha.pathTo, System.StringComparison.OrdinalIgnoreCase) == 0)
+                    {
+                        cha.Status = GitChangeInfo.ChangeType.MODIFY;
+                    }
+                }
+            }
+
+            var deletes = remainingChanges.Where(cha => cha.Status == GitChangeInfo.ChangeType.DELETE).ToArray();
+            var adds = remainingChanges.Where(cha => cha.Status == GitChangeInfo.ChangeType.ADD).ToArray();
+            var removeElement = "RemoveElement";
+            foreach (var cha in remainingChanges)
+            {
+                //filter deletes out, which are really case only  renames
+                if (cha.Status == GitChangeInfo.ChangeType.DELETE)
+                {
+                    if (adds.Any(add => String.Equals(add.path, cha.path, System.StringComparison.OrdinalIgnoreCase)))
+                    {
+                        cha.Status = removeElement;
+                    }
+                }
+
+                //change adds to renameedit, if the file name is the same as a delete
+                if (cha.Status == GitChangeInfo.ChangeType.ADD)
+                {
+                    if (deletes.Any(del => String.Equals(cha.path, del.path, System.StringComparison.OrdinalIgnoreCase)))
+                    {
```

Comment:
For efficiency and readability, isn't it possible to that at the same time we handle the `DELETE` change (line 90)?
We already search for the `ADD` change corresponding to the `DELETE`. We just have to update it at the same time.
This way we prevent 2 more searches (and remove most of this code)....